tests/feature/providers/test-vip-provider.php

```diff
@@ -0,0 +1,127 @@
+<?php
+namespace Alley\WP\WP_Page_Cache_Control\Tests\Feature\Providers;
+
+use Alley\WP\WP_Page_Cache_Control\Header;
+use Alley\WP\WP_Page_Cache_Control\Providers\VIP_Provider;
+use Alley\WP\WP_Page_Cache_Control\Tests\Test_Case;
+use Automattic\VIP\Cache\Vary_Cache;
+use WPCOM_VIP_Cache_Manager;
+
+/**
+ * WordPress VIP Provider
+ */
+class Test_VIP_Provider extends Test_Case {
+	protected function setUp(): void {
+		parent::setUp();
+
+		add_filter( 'wp_page_cache_control_provider', fn () => VIP_Provider::class );
+
+		$this->assertInstanceOf( VIP_Provider::class, wp_page_cache_control() );
+	}
+
+	protected function tearDown(): void {
+		parent::tearDown();
+
+		Vary_Cache::unload();
+
+		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+		unset( $_SERVER['HTTP_TRUE_CLIENT_IP'], $_SERVER['HTTP_X_FORWARDED_FOR'], $_SERVER['HTTP_USER_AGENT'] );
+	}
+
```

Comment:
Would it make sense to store the global $_SERVER variable so we don't interact with it directly? Not sure if there are potential side effects to doing so. 

```php
protected function setUp(): void {
        parent::setUp();
        $this->_server_original = $_SERVER;  // Store the original $_SERVER variable

        // Rest of the setup continues...
    }

    protected function tearDown(): void {
        // Restore the original $_SERVER variable.
        $_SERVER = $this->_server_original;

        parent::tearDown();

        // Vary_Cache unloading remains...
    }
```